spring-cloud-stream-module-launcher/src/main/java/org/springframework/cloud/stream/module/launcher/ModuleLauncher.java

```diff
@@ -62,23 +62,27 @@ public ModuleLauncher(ModuleResolver moduleResolver) {
 	 * The format of each module must conform to the <a href="http://www.eclipse.org/aether">Aether</a> convention:
 	 * <code>&lt;groupId&gt;:&lt;artifactId&gt;[:&lt;extension&gt;[:&lt;classifier&gt;]]:&lt;version&gt;</code>
 	 *
-	 * To pass arguments to a module, prefix with the module name and a dot. The arg name will be de-qualified and passed along.
-	 * For example: <code>--org.springframework.cloud.stream.module:time-source:1.0.0.BUILD-SNAPSHOT.bar=123</code> becomes <code>--bar=123</code> and is only passed to the 'org.springframework.cloud.stream.module:time-source:1.0.0.BUILD-SNAPSHOT' module.
-	 *
-	 * @param modules a list of modules
-	 * @param args a list of arguments, prefixed with the module name
+	 * @param moduleLaunchRequests a list of modules with their (unqualified) arguments
+	 */
+	public void launch(Iterable<ModuleLaunchRequest> moduleLaunchRequests) {
+		for (ModuleLaunchRequest moduleLaunchRequest : moduleLaunchRequests) {
+			String module = moduleLaunchRequest.getModule();
+			moduleLaunchRequest.addArgument("spring.jmx.default-domain", module.replace("/", ".").replace(":", "."));
+			launchModule(module, toProgramArguments(moduleLaunchRequest.getArguments()));
+		}
+	}
+
+	/**
+	 * Converts a set of semantic program arguments to "command line program arguments" that is, to the
+	 * {@literal --foo=bar} form.
 	 */
-	public void launch(String[] modules, String[] args) {
-		for (String module : modules) {
-			List<String> moduleArgs = new ArrayList<>();
-			for (String arg : args) {
-				if (arg.startsWith("--" + module + ".")) {
-					moduleArgs.add("--" + arg.substring(module.length() + 3));
-				}
-			}
-			moduleArgs.add("--spring.jmx.default-domain=" + module.replace("/", ".").replace(":", "."));
-			launchModule(module, moduleArgs.toArray(new String[moduleArgs.size()]));
+	private String[] toProgramArguments(Map<String, String> args) {
```

Comment:
I know this is a private method, but I would probably rename "toArgArray" (the term "program" feels out of context).